### PR TITLE
Add support for embeding bitcode as clang's -fembed-bitcode flag does. 

### PR DESCRIPTION
**Pull request by zvookin**
The main use for this is to satisfy Apple's requirement for bitcode inclusion for watchOS and tvOS app submission.

Two questions here are what to use for the command line and whether the bitcode should be written pre-optimization or post.

### CODE REVIEW

File: src/CodeGen_Internal.cpp [hunk 1]
```diff
@@ -476,5 +476,82 @@ void set_function_attributes_for_target(llvm::Function *fn, Target t) {
     fn->addFnAttr("reciprocal-estimates", "none");
 }
 
+void embed_bitcode(llvm::Module *M, const string &halide_command) {
```
Comment (steven-johnson) [inline src/CodeGen_Internal.cpp]: I know that this code is meant to emulate the ` -fembed-bitcode` flag of Clang; just curious if the necessary behavior is actually documented anywhere, vs just reverse-engineered from Clang. Probably worth adding some comments about that either way, with link to spec or comments about where in Clang the behavior was observed, in order to make future maintenance simpler.

**Comment by steven-johnson** [comment]
Is this ready to land?

**Comment by steven-johnson** [comment]
What's the latest on this? IIRC we got an update in the Gitter channel but it didn't make it into comments here.

**Comment by shoaibkamil** [comment]
Still working with Apple engineers to figure out how to test this.

**Pull request merged**